src/ir_Daikin.cpp

```diff
@@ -261,25 +268,457 @@ void IRDaikinESP::setSwingHorizontal(bool state) {
+
+  if (value & 0b00000000000000001000000000000000)
+    setEcono(true);
+  if (value & 0b00000000000000010000000000000000)
+    setPowerful(true);
+  if (value & 0b00000000000000100000000000000000)
+    setQuiet(true);
+  if (value & 0b00000000000001000000000000000000)
+    setSensor(true);
+  if (value & 0b00000000000010000000000000000000)
+    setSwingVertical(true);
+
+  value = value >> 20;
+  setCurrentTime(value);
+}
+
 #endif  // SEND_DAIKIN
 
 #if DECODE_DAIKIN
-// TODO(crankyoldgit): NOT WORKING. This needs to be finished.
-// Decode the supplied Daikin A/C message. (NOT WORKING - DO NOT USE)
+
+#define DAIKIN_CURBIT DAIKIN_COMMAND_LENGTH
+#define DAIKIN_CURINDEX DAIKIN_COMMAND_LENGTH+1
+#define OFFSET_ERR 65432
+
+#define DAIKIN_TOLERANCE 35
+#define DAIKIN_MARK_EXCESS MARK_EXCESS
+
+void addbit(bool val, unsigned char data[]) {
+  uint8_t curbit = data[DAIKIN_CURBIT];
+  uint8_t curindex = data[DAIKIN_CURINDEX];
+  if (val) {
+    unsigned char bit = 1;
+    bit = bit << curbit;
+    data[curindex] |= bit;
+  }
+  curbit++;
+  if (curbit == 8) {
+    curbit = 0;
+    curindex++;
+  }
+  data[DAIKIN_CURBIT] = curbit;
+  data[DAIKIN_CURINDEX] = curindex;
+}
+
+uint16_t checkheader(decode_results *results, uint16_t offset, unsigned char daikin_code[]) {
+  if (!IRrecv::matchMark(results->rawbuf[offset++], DAIKIN_ZERO_MARK,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+    return OFFSET_ERR;
+  if (!IRrecv::matchSpace(results->rawbuf[offset++], DAIKIN_ZERO_SPACE + DAIKIN_GAP,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+    return OFFSET_ERR;
+  if (!IRrecv::matchMark(results->rawbuf[offset++], DAIKIN_HDR_MARK,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+    return OFFSET_ERR;
+  if (!IRrecv::matchSpace(results->rawbuf[offset++], DAIKIN_HDR_SPACE,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+    return OFFSET_ERR;
+  /*
+  if (daikin_code[DAIKIN_CURBIT] != 0) {
+    daikin_code[DAIKIN_CURBIT] = 0;
+    daikin_code[DAIKIN_CURINDEX]++;
+  }
+  */
+  return offset;
+}
+
+uint16_t readbits(decode_results *results, uint16_t offset, unsigned char daikin_code[], uint16_t countbits) {
+  for (uint16_t i = 0; i < countbits && offset < results->rawlen-1; i++, offset++) {
+    if (!IRrecv::matchMark(results->rawbuf[offset++], DAIKIN_ONE_MARK,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+      return OFFSET_ERR;
+    if (IRrecv::matchSpace(results->rawbuf[offset], DAIKIN_ONE_SPACE,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+      addbit(1, daikin_code);
+    else if (IRrecv::matchSpace(results->rawbuf[offset], DAIKIN_ZERO_SPACE,
+              DAIKIN_TOLERANCE, DAIKIN_MARK_EXCESS))
+      addbit(0, daikin_code);
+    else
+      return OFFSET_ERR;
+  }
+  return offset;
+}
+
+
+// Decode the supplied Daikin A/C message.
 // Args:
 //   results: Ptr to the data to decode and where to store the decode result.
 //   nbits:   Nr. of bits to expect in the data portion. Typically SAMSUNG_BITS.
```

Comment:
Oops. Looks like I left SAMSUNG_BITS. Can you please change it to DAIKIN_BITS? Thanks.